src/controllers/productController.js

```diff
@@ -94,11 +94,10 @@ class productController{
 
   static async deleteProduct(req,res){
     try {
-      if (req.user.role.roleNam !== "vendor") {
-        console.log(req.user.role.roleName);
+      if (req.user.roleName!== "vendor") {
```

Comment:
Same as here. To avoid repeating this code and make it more reusable, would you consider creating a middleware function that validates the user's role and returns an error response if the user is not authorized? This way, you can simply call the middleware function in each of the relevant routes and avoid duplicating the same code in multiple places.